drf_s3_direct/api/views.py

```diff
@@ -0,0 +1,174 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import decorators, response, status, viewsets
+from rest_framework.permissions import BasePermission
+from rest_framework.request import Request
+
+from .. import services
+from . import permissions, serializers
+
+
+class S3FileView(viewsets.GenericViewSet):
+    """View for getting params for s3."""
+
+    file_path_regex = (
+        r"(?P<app>\w+)/(?P<model>\w+)/(?P<pk>[\w-]+)/(?P<field>\w+)"
+    )
+
+    serializer_class = serializers.S3RequestParamsSerializer
+
+    @decorators.action(
+        methods=["POST"],
+        url_path="get-params",
+        url_name="get-params",
+        detail=False,
+    )
+    def get_params(self, request: Request) -> response.Response:
+        """Get parameters for upload to S3 bucket.
+
+        Current endpoint returns all required for s3 upload data,
+        which should be later sent to `url` as `form-data` url with
+        'file'. Workflow: First, you make request to this endpoint. Then send
+        response data to `url` via `POST`.
+
+        """
+        serializer = serializers.S3RequestParamsSerializer(
+            context_request=request,
+            data=request.data,
+        )
+        serializer.is_valid(raise_exception=True)
+        params = services.generate_params(
+            user=request.user,
+            **serializer.data,
+        )
+        return response.Response(
+            status=status.HTTP_200_OK,
+            data=serializers.S3UploadSerializer(
+                instance=dataclasses.asdict(params),
+            ).data,
+        )
+
+    @decorators.action(
+        url_path=f"check-file-access/{file_path_regex}",
+        detail=False,
+    )
+    def check_file_access(
+        self,
+        request: Request,
+        app: str,
+        model: str,
+        pk: str,
+        field: str,
+    ) -> response.Response:
+        """Check if user has access to file."""
+        instance = self.get_instance(
+            app=app,
+            model=model,
+            pk=pk,
+        )
+        if not self.check_access_permissions(
+            instance=instance,
+            field=field,
+        ):
+            raise Http404
+        file_field: models.FieldFile | None = getattr(instance, field, None)
+        if not file_field:
+            raise Http404
+        return response.Response(status=status.HTTP_200_OK)
+
+    @decorators.action(
+        url_path=f"get-file/{file_path_regex}",
+        detail=False,
+    )
+    def get_file(
+        self,
+        request: Request,
+        app: str,
+        model: str,
+        pk: str,
+        field: str,
+    ) -> response.Response:
+        """Get download link."""
+        instance = self.get_instance(
+            app=app,
+            model=model,
+            pk=pk,
+        )
+        if not self.check_access_permissions(
+            instance=instance,
+            field=field,
+        ):
+            raise Http404
```

Comment:
```suggestion
            raise NotFound
```
Maybe [NotFound](https://www.django-rest-framework.org/api-guide/exceptions/#notfound) from DRF more suitable

Reply:
![image](https://github.com/saritasa-nest/drf-s3-direct/assets/43534015/6a9ac30f-2a36-4042-8cc9-aa6d2aea0874)